src/vs/workbench/api/test/browser/extHostTelemetry.test.ts

```diff
@@ -0,0 +1,129 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { URI } from 'vs/base/common/uri';
+import { ExtensionIdentifier, IExtensionDescription, TargetPlatform } from 'vs/platform/extensions/common/extensions';
+import { DEFAULT_LOG_LEVEL } from 'vs/platform/log/common/log';
+import { TelemetryLevel } from 'vs/platform/telemetry/common/telemetry';
+import { TestTelemetryLoggerService } from 'vs/platform/telemetry/test/common/telemetryLogAppender.test';
+import { IExtHostInitDataService } from 'vs/workbench/api/common/extHostInitDataService';
+import { ExtHostTelemetry } from 'vs/workbench/api/common/extHostTelemetry';
+import { IEnvironment } from 'vs/workbench/services/extensions/common/extensionHostProtocol';
+import { mock } from 'vs/workbench/test/common/workbenchTestServices';
+import type { TelemetryAppender } from 'vscode';
+
+suite('ExtHostTelemetry', function () {
+
+	const mockEnvironment: IEnvironment = {
```

Comment:
Is this the suggested way to do things @jrieken? Just create these bogus environment objects and Extension identifiers? I don't need most of these properties, but this seemed like the correct way to do it. Let me know if there's a better way to do things :)

Reply:
Yeah, I like to mock/fake as much as possible so that my dependencies stay small. Also note that we have the [mock](https://github.com/microsoft/vscode/blob/4674509f002f98319f48e44460561c85b3d8ad15/src/vs/base/test/common/mock.ts#L12)-util which makes it super easy to mock and/or partially implement any service or dependency